docs/source/reference/modding/extended.rst

```diff
@@ -0,0 +1,208 @@
+Extended modding feature overview
+#################################
+
+OpenMW supports some extended moddong features out of box.
+These features can conflict with some mods, and such situations should be handled as any mod other conflict - a conflicted mod should be patched.
+The engine itself does not provide any kind of blacklisting.
+
+
+Native herbalism support
+------------------------
+
+In OpenMW it is possible to add one or many NiSwitchNodes with a ``HerbalismSwitch`` name.
+Every such switch should have at least two child nodes (a first one represents the node in the normal mode, the second one - in the harvested mode).
+If an organic container's mesh has such nodes, it it considered as a plant. During activation a window with plant's content is not shown,
+OpenMW moves its content directly to player, triggers a theft crime if the plant is owned and switches its mesh to the harvested state.
+
+Also it is possible to use scripts with OnActivate command for such containers. For example, when player needs a tool to harvest a plant (e.g. a pickaxe for ore).
+
+Keep in mind that a collision detection system does not consider switch nodes, so use the RootCollisionNode or the ``NCO`` NiStringExtraData for harvestable meshes.
+
+Advantages of described approach over old herbalism mods:
+
+1. There is no need to spawn separate "harvested" objects
+
+2. There is no need to attach a script to every container
+
+3. It supports an ownership check (an original engine without MWSE does not)
+
+4. It does not alter an original respawn mechanics
+
+An example of mod which uses this feature is the `Graphic Herbalism`_.
+
+Animated containers support
+---------------------------
+
+It is possible to attach opening/closing animations for containers. To do it, you need to create a KF-file for container with such groups:
+
+1. ``Idle`` (with ``Start`` and ``Stop`` keys)
+
+2. ``ContainerOpen`` (with ``Start``, ``Loot`` and ``Stop`` keys)
+
+3. ``ContainerClose`` (with ``Start`` and ``Stop`` keys)
+
+The ``Idle`` group can have the zero time, the ``Loot`` key for ``ContainerOpen`` allows to play a part of opening animation in the background.
+
+For example, with such setup the opening animation has the 1.0 sec duration, but shows the container window in the middle of opening animation:
+
+``0.0: ContainerOpen: start``
+
+``0.5: ContainerOpen: loot``
+
+``1.0: ContainerOpen: stop``
+
+Also it is possible to attach opening/closing sounds to container's animations:
+
+``1.0: ContainerClose: open``
+
+``1.01: Sound: AC_dw_drawer_close``
+
+``2.0: ContainerClose: stop``
+
+An approach is the same as for other animation sounds (you need to register sound record in the ESM/ESP file).
+Note that the sound time is slightly higher than the closing animation start, otherwise sound will be played in the end of opening animation in this example.
```

Comment:
```suggestion
Note that the sound starting time value is slightly higher than the closing animation start, otherwise sound will be played at the end of opening animation in this example.
```